Sources/EmbraceCoreDataInternal/BackgroundTaskWrapper.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  Copyright © 2025 Embrace Mobile, Inc. All rights reserved.
+//
+
+import Foundation
+
+#if canImport(UIKit) && !os(watchOS)
+import UIKit
+#if !EMBRACE_COCOAPOD_BUILDING_SDK
+import EmbraceCommonInternal
+#endif
+
+/// This class is a wrapper around `UIApplication.shared.beginBackgroundTask`.
+/// Based off https://developer.apple.com/forums/thread/85066 and https://developer.apple.com/forums/thread/729335
+class BackgroundTaskWrapper {
+
+    let name: String
+    private var taskID: UIBackgroundTaskIdentifier
+
+    init?(name: String, logger: InternalLogger) {
+        self.name = name
+        self.taskID = .invalid
+
+        let taskID = UIApplication.shared.beginBackgroundTask(withName: name) {
+            logger.critical("Background task \(name) cancelled!")
```

Comment:
This is actually "expiration". You have a very limited amount of time to cancel your work here and end the task. I don't see anything around ensuring the "task" that is being wrapped in this gets notified to cancel, so basically things will still crash in some (probably most) circumstances.